### PR TITLE
feat: qr code login option

### PR DESCRIPTION
**Pull request by kvcache**
`momento login qr`

Prints a unicode QR image for you to log in via your phone.
`momento login` defaults to `momento login browser`.

### CODE REVIEW

File: src/commands/login.rs [hunk 1]
```diff
@@ -18,6 +35,25 @@ pub async fn login() -> LoginResult {
...
         }
-    })
-    .await
+    }
+}
+
+fn login_with_qr_code(action: LoginAction) -> EarlyOutActionResult {
+    match action {
+        momento::momento::auth::LoginAction::OpenBrowser(open) => {
+            eprintln!("Navigate here to log in: {}", open.url);
```
Comment (tylerburdsall) [inline src/commands/login.rs:44]: Shouldn't this be `println!`? I thought `eprintln` was only for `stderr`. Or is this some workaround to get the QR code to actually show up?

File: src/commands/login.rs [hunk 2]
```diff
@@ -18,6 +35,25 @@ pub async fn login() -> LoginResult {
...
+            let code = QrCode::new(open.url).unwrap();
+            let image = code
+                .render::<unicode::Dense1x2>()
+                .dark_color(unicode::Dense1x2::Dark)
+                .light_color(unicode::Dense1x2::Light)
+                .build();
+            eprintln!("{}", image);
```
Comment (tylerburdsall) [inline src/commands/login.rs:51]: Non-blocking, curious to know how this would appear on light-theme terminal background

**Review by tylerburdsall — COMMENTED** [review]
Just a couple comments, everything is looking good though!

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Review by tylerburdsall — APPROVED** [review]

File: src/commands/login.rs (hunk 2 above)
Comment (kvcache) [reply src/commands/login.rs:51]: image

Comment (tylerburdsall) [reply src/commands/login.rs:51]: Nice! Cool to see that it handles that

**Pull request merged**